test_runner/regress/test_logical_replication.py

```diff
@@ -221,6 +221,37 @@ def slot_removed(ep):
     wait_until(number_of_iterations=10, interval=2, func=partial(slot_removed, endpoint))
 
 
+def test_ondemand_wal_download_in_replication_slot_funcs(neon_env_builder: NeonEnvBuilder):
+    neon_env_builder.num_safekeepers = 3
+    env = neon_env_builder.init_start()
+
+    env.neon_cli.create_branch("init")
+    endpoint = env.endpoints.create_start("init")
+
+    with endpoint.connect().cursor() as cur:
+        cur.execute("create table wal_generator (id serial primary key, data text)")
+        cur.execute(
+            """
+INSERT INTO wal_generator (data)
+SELECT repeat('A', 1024) -- Generates a kilobyte of data per row
+FROM generate_series(1, 16384) AS seq; -- Inserts enough rows to exceed 16MB of data
+"""
+        )
+        cur.execute("create table t(a int)")
+        cur.execute(
+            "SELECT * FROM pg_create_logical_replication_slot('slotty_mcslotface', 'test_decoding')"
```

Comment:
I don't see what is the point of generating a segment of WAL data if slot is created later; pg_logical_slot_peek_binary_changes won't stream these 16MB anyway, so for this to make sense slot should be created before bulk insertion.

But in principle just restart of compute is enough to test on demand download as current segment on the new compute would be empty.

Reply:
Good point, I moved it above. Test still seems to pass